src/commands/test/test_command_test.py

```diff
@@ -0,0 +1,59 @@
+# pylint: disable=invalid-name
+import re
+from dataclasses import fields
+from pathlib import Path
+from typing import cast
+from unittest.mock import MagicMock
+
+import pytest
+from pytest_mock import MockerFixture
+
+from src.commands.test import TestCommand
+
+
+@pytest.mark.asyncio
+async def test_test_command_runs_scheduler_properly(mocker: MockerFixture):
+    args = TestCommand.Args(
+        target=Path("foo"),
+        match=re.compile("foo"),
+        omit=re.compile("bar"),
+        cairo_path=[Path() / "baz"],
+    )
+    TestCollectorMock = mocker.patch(
```

Comment:
Why do we need to mock here 🤔

Reply:
This way we don't even check if correct arguments are passed, much much better would be to let scheduler run some test

Reply:
I don't want to mix unit and integration tests if it's possible. I'll use `assert_called_once_with` to check the args.
Running actual tests will make tests slower, and it won't check if all arguments are passed properly.

Reply:
I thought we are testing TestScheduler here, my mistake 😅